app/src/main/java/org/openobservatory/ooniprobe/activity/MeasurementDetailActivity.java

```diff
@@ -129,7 +130,9 @@ public static Intent newIntent(Context context, int id) {
 		switch (item.getItemId()) {
 			case R.id.rawData:
 				try {
-					String json = FileUtils.readFileToString(Measurement.getEntryFile(this, measurement.id, measurement.test_name), StandardCharsets.UTF_8);
+					File entryFile = Measurement.getEntryFile(this, measurement.id, measurement.test_name);
+					entryFile.mkdirs();
```

Comment:
Same here. You should not be attempting to create the directory when you are reading it.

Also by calling `mkdirs()` on a file you are creating a directory which has the name of a file which I don't think is what you want.

Reply:
I was mistaken, `mkdirs()` actually creates up to the parent directory when operating on a file. 

The comment on not creating the parent directory when you are trying to read the file though still holds true.